halo2_proofs/examples/serialization.rs

```diff
@@ -48,6 +48,15 @@ impl StandardPlonkConfig {
 
         [a, b, c].map(|column| meta.enable_equality(column));
 
+        // TEST for "compress_selectors" logic correctness
+        let s = meta.selector();
+        meta.create_gate("selector gate", |meta| {
+            let s = meta.query_selector(s);
+            let a = meta.query_advice(a, Rotation::cur());
+            let b = meta.query_advice(b, Rotation::cur());
+            vec![s * (a - b)]
+        });
+
```

Comment:
I would consider removing this.  The reason is that this circuit is called `StandardPlonk`, and this gate is not part of the standard plonk; so as an example it can be confusing.

Moreover, I don't see any update on the example that assigns values to the new selector `s`.

I think it's better to have a unit test for selector to fixed checks (both compression and without, different number of selectors, different expression degree, etc.)

Reply:
@ed255 
I add the unit tests for `compress_selectors`.
Pls check this [file](https://github.com/privacy-scaling-explorations/halo2/pull/322/files#diff-28193f64f728006ed49fec5b433b483eca736a93165803e30edd541f1793907e) if it is able to cover all possible cases.